src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs

```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                for (var i = 0; i < imports.Count; i++)
+                {
+                    if (string.Equals(imports[i].Source.FileName, node.Source.Value.FilePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is actually doing it the hard way. You already have the source document, and the source document has a file name. You just want to match on `!=` on the source document.